Library/ViewModels/CommentCellViewModel.swift

```diff
@@ -51,32 +54,49 @@ public final class CommentCellViewModel:
       Format.date(secondsInUTC: $0.createdAt, dateStyle: .medium, timeStyle: .short)
     }
 
-    let badge = self.commentUser.signal.skipNil()
-      .map { comment, user in
+    let badge = self.commentUserAndProject.signal.skipNil()
+      .map { comment, user, _ in
         comment.author.id == user?.id.description ? .you : comment.authorBadge
       }
 
     self.authorBadge = Signal.merge(
       badge,
       badge.takeWhen(self.bindStylesProperty.signal)
     )
+
+    let isLoggedOut = self.commentUserAndProject.signal
+      .ignoreValues()
+      .map { _ in AppEnvironment.current.currentUser }
+      .map(isNil)
+
+    let isBacker = self.commentUserAndProject.signal
+      .skipNil()
+      .map { _, _, project in project }
+      .map(userIsBackingProject)
+
+    self.replyButtonIsHidden = Signal.combineLatest(isLoggedOut, isBacker)
+      .map { isLoggedOut, isBacker in
+        // If the user is logged out OR logged in AND not a backer, hide replyButton
+        isLoggedOut || (!isLoggedOut && isBacker == false)
```

Comment:
what about:
```swift
guard isLoggedOut else {
  return !isBacker
}

return true
```
or:
```swift
let isNotABacker = self.commentUserAndProject.signal
  .skipNil()
  .map { _, _, project in project }
  .map(userIsBackingProject)
  .negate()

self.replyButtonIsHidden = Signal.combineLatest(isLoggedOut, isNotABacker)
  .map { isLoggedOut, isNotABacker in isLoggedOut || isNotABacker }
```

Reply:
Could be `loggedOut || !isBacker`

Reply:
Nice, going with the latter. I think it reads well and I can delete the comment I left above it, as a result.